### PR TITLE
Add check for non-existent parent ID to AddSubMenuRequest

### LINKED ISSUE #3524: SDL transfers AddSubMenu request with non-existent parentID parameter to HMI
### Bug Report
SDL transfers AddSubMenu request with non-existent parentID parameter to HMI

#### Preconditions:

1. SDL and HMI are started
2. Mobile device connected
3. Mobile app is registered and activated

##### Reproduction Steps
1. Add submenu item AddSubMenu RPC with non-existent parentID parameter
AddSubMenu(parentID=1) 

##### Expected Behavior
APP→SDL: AddSubMenu ( "menuName":"Submenu Name 1","parentID":1, "menuID":1 )
SDL→APP: AddSubMenu ( resultCode: INVALID_ID, success=false ).

##### Observed Behavior
APP→SDL: AddSubMenu ( "menuName":"Submenu Name 1","parentID":1, "menuID":1 )
SDL→HMI: UI.AddSubMenu ( params, "menuParams":{"menuName":"Submenu Name 1","parentID":1,"position":0}, "menuID":1} )

##### OS & Version Information
SDL: https://github.com/smartdevicelink/sdl_core/commits/develop (9f61819)
HMI: https://github.com/smartdevicelink/sdl_hmi/tree/develop (f712d4e)

##### Test Case, Sample Code, and / or Example App
Logs:[SDL_logs.zip](https://github.com/smartdevicelink/sdl_core/files/5313925/SDL_logs.zip)

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #3524
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Covered by ATF test script.
New unit test is added.

### Summary
Because SDL transfers AddSubMenu request with non-existent parentID parameter to HMI, there is a need in additional check for presence of this parentID in SubMenu map. That's why this check is added to AddSubMenuRequest::Run() method.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by ychernysheva** [comment]
@theresalech This fix is ready for review.
Thank you!

**Comment by theresalech** [comment]
@ychernysheva thank you for letting us know. We'll review and work to include a fix in the second release candidate for Core 7.0 (to be delivered to the Steering Committee on 11/2).

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/add_sub_menu_request.cc [hunk 1]
```diff
@@ -96,14 +96,23 @@ void AddSubMenuRequest::Run() {
...
   const uint32_t parent_id =
       received_msg_params.keyExists(strings::parent_id)
           ? received_msg_params[strings::parent_id].asUInt()
           : 0;
 
+  if (0 != parent_id) {
+    smart_objects::SmartObject parent = app->FindSubMenu(parent_id);
+    if (smart_objects::SmartType_Null == parent.getType()) {
+      SDL_LOG_ERROR("Parent ID " << parent_id << " doesn't exist");
+      SendResponse(false, mobile_apis::Result::INVALID_ID);
```
Comment (iCollin) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/add_sub_menu_request.cc:108]: did you consider adding a message to this response to explain why invalid id was returned?

Comment (ychernysheva) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/add_sub_menu_request.cc]: @iCollin Reason is added in 017d900f74ba6e8c573e13dbed257d437af70312

**Review by iCollin — APPROVED** [review]

**Pull request merged**